backend/src/middlewares/userValidation.js

```diff
@@ -0,0 +1,43 @@
+const { z } = require("zod");
+
+const userSchema = z.object({
+  firstname: z
+    .string()
+    .min(2, { message: "Firstname must contain at least 2 characters" })
```

Comment:
When you need a specific message depending on value prefer using the syntax :
min({value :2, message: "your message"}). Easier to read, so easier to maintain